lib/cli/ui/frame.rb

```diff
@@ -225,21 +257,31 @@ def edge(text, color: raise, first: raise, right_text: nil)
             prefix << ' ' << text << ' '
           end
 
-          termwidth = CLI::UI::Terminal.width
+          termwidth = CLI::UI::Terminal.width if frame_style == :box
 
           suffix = +''
+
           if right_text
-            suffix << ' ' << right_text << ' '
-          end
+            case frame_style
+            when :box
+              suffix << ' ' << right_text << ' '
+            when :bracket
+              prefix << ' ' << right_text << ' '
+            end
 
-          suffix_width = CLI::UI::ANSI.printing_width(suffix)
-          suffix_end   = termwidth - 2
-          suffix_start = suffix_end - suffix_width
+          end
 
           prefix_width = CLI::UI::ANSI.printing_width(prefix)
           prefix_start = 0
           prefix_end   = prefix_start + prefix_width
 
+          # Artificially shrink the calculated width of the terminal to just the length of the prefix
+          termwidth = prefix_end if frame_style == :bracket
```

Comment:
This feels too much like cheating. How hard is it to bypass all of that cursor arithmetic garbage with bracket mode rather than tricking it like this?

I wonder if we should just split this method into two paths rather than trying to conditionalize it? I imagine `edge_bracket` would be waaaayyy simpler.

Reply:
I just did a bit of surgery on the method and it the first 12 lines are all basically the same but the rest seems trivial:

```ruby
        def edge(text, color: raise, first: raise, right_text: nil)
          color = CLI::UI.resolve_color(color)
          text  = CLI::UI.resolve_text("{{#{color.name}:#{text}}}")

          prefix = +''
          FrameStack.items.each do |item|
            prefix << CLI::UI.resolve_color(item).code << CLI::UI::Box::Heavy::VERT
          end
          prefix << color.code << first << (CLI::UI::Box::Heavy::HORZ * 2)
          text ||= ''
          unless text.empty?
            prefix << ' ' << text << ' '
          end

          suffix = right_text ? " #{right_text}" : ''
          color.code + prefix + color.code + suffix + CLI::UI::Color::RESET.code + "\n"
        end
```